wisp/wisp-rate-limiting/src/main/kotlin/wisp/ratelimiting/RateLimiter.kt

```diff
@@ -27,13 +27,37 @@ interface RateLimiter {
     amount: Long = 1
   ): ConsumptionData
 
+  /**
+   * This tests whether [amount] tokens are available in the bucket associated with the given key.
+   * It is essentially a dry run of [consumeToken].
+   * Note that this data may be stale when it comes back, as time has elapsed and other pods
+   * could have taken tokens in the meantime.
+   */
+  fun testConsumptionAttempt(
```

Comment:
What’s the use case for this?

Reply:
Essentially the same as `availableTokens` but it provides more information, namely the timestamp when the bucket will be refilled. Callers might want to know this if there weren't enough tokens in the bucket for their test consumption